ui/src/types/alerting.ts

```diff
@@ -1,9 +1,58 @@
+import {StatusRule, NotificationRuleBase, TagRule} from 'src/client'
+
+export interface AddID<T> {
+  id: string
+  value: T
+}
+
+export type StatusRuleItem = AddID<StatusRule>
+export type TagRuleItem = AddID<TagRule>
+
+type ExcludeKeys<T> = Pick<T, Exclude<keyof T, 'statusRules' | 'tagRules'>>
+
+export interface NotificationRuleBaseBox
+  extends ExcludeKeys<NotificationRuleBase> {
+  schedule: 'cron' | 'every'
+  statusRules: StatusRuleItem[]
+  tagRules: TagRuleItem[]
+}
+
+export type NotificationRuleBox = SlackRule | SMTPRule | PagerDutyRule
+
+export type SlackBase = {
```

Comment:
Should these be part of the swagger spec instead?

Reply:
Yuppers!

Reply:
Going to do this in another PR as it looks like swagger was recently updated and there are about a bagillion errors now.